### PR TITLE
repoinfo: Implement JSON output

### PR DESCRIPTION
**Pull request by jan-kolarik**
Turn on `quiet` mode when `--json` is requested. Also fixed suppressing repository download progress when `quiet` mode is turned on.

For: #867.

### CODE REVIEW

File: libdnf5-cli/output/repo_info.cpp [hunk 1]
```diff
@@ -22,10 +22,25 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "key_value_table.hpp"
 #include "utils/string.hpp"
 
+#include "libdnf5-cli/tty.hpp"
 #include "libdnf5-cli/utils/units.hpp"
 
+#include <json.h>
```
Comment (kontura) [inline libdnf5-cli/output/repo_info.cpp:28]: This is a nitpick but I would prefer to include as little as possible, only: `json-c/json_object.h` should suffice but feel free to leave it as it is.

File: libdnf5-cli/output/repo_info.cpp [hunk 2]
```diff
@@ -22,10 +22,25 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "key_value_table.hpp"
 #include "utils/string.hpp"
 
+#include "libdnf5-cli/tty.hpp"
```
Comment (kontura) [inline libdnf5-cli/output/repo_info.cpp:25]: Can I ask why not to include directly `<iostream>`?

File: libdnf5-cli/output/repo_info.cpp (hunk 1 above)
Comment (jan-kolarik) [reply libdnf5-cli/output/repo_info.cpp:28]: Yep, agree.

File: libdnf5-cli/output/repo_info.cpp (hunk 2 above)
Comment (jan-kolarik) [reply libdnf5-cli/output/repo_info.cpp:25]: Hmm, just a leftover, I've initially used something from that `tty` header, will change that...

**Comment by kontura** [comment]
Thank you.

**Comment by kontura** [comment]
Oh I also forgot to mention, I think it would be good to have some tests for the `--json` option.
Plus it would be nice to have a test for the `--refresh --quiet` combo.

**Comment by jan-kolarik** [comment]
Yep, sure. Thanks for the notice. I will create a follow-up ticket for that.

**Pull request merged**